src/EnergyPlus/UnitVentilator.cc

```diff
@@ -804,469 +650,364 @@ namespace UnitVentilator {
+                                ShowContinueError(state, "...specified in " + CurrentModuleObject + "=\"" + unitVent.Name + "\".");
                                 ErrorsFound = true;
                             }
                         }
                     }
 
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).MinVolColdWaterFlow = 0.0;
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).ColdControlOffset = Numbers(5);
+                    unitVent.MinVolColdWaterFlow = 0.0;
+                    unitVent.ColdControlOffset = Numbers(5);
                     // Set default convergence tolerance
-                    if (state.dataUnitVentilators->UnitVent(UnitVentNum).ColdControlOffset <= 0.0) {
-                        state.dataUnitVentilators->UnitVent(UnitVentNum).ColdControlOffset = 0.001;
+                    if (unitVent.ColdControlOffset <= 0.0) {
+                        unitVent.ColdControlOffset = 0.001;
                     }
                     {
-                        auto const SELECT_CASE_var(state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilType);
-
-                        if (SELECT_CASE_var == Cooling_CoilWaterCooling) {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).MaxVolColdWaterFlow = GetWaterCoilMaxFlowRate(
-                                state, "Coil:Cooling:Water", state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilName, ErrorsFound);
-                        } else if (SELECT_CASE_var == Cooling_CoilDetailedCooling) {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).MaxVolColdWaterFlow =
-                                GetWaterCoilMaxFlowRate(state,
-                                                        "Coil:Cooling:Water:DetailedGeometry",
-                                                        state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilName,
-                                                        ErrorsFound);
-                        } else if (SELECT_CASE_var == Cooling_CoilHXAssisted) {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).MaxVolColdWaterFlow =
-                                GetHXAssistedCoilFlowRate(state,
-                                                          "CoilSystem:Cooling:Water:HeatExchangerAssisted",
-                                                          state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilName,
-                                                          ErrorsFound);
-                        } else {
+                        switch (unitVent.CCoilType) {
+                        case CoolCoilType::Water: {
+                            unitVent.MaxVolColdWaterFlow =
+                                WaterCoils::GetCoilMaxWaterFlowRate(state, "Coil:Cooling:Water", unitVent.CCoilName, ErrorsFound);
```

Comment:
The type argument should be an `enum` here, not a `string`. Alternatively, if you know the `CoilIndex`, you should just use that instead of having to look for it.

Reply:
These legacy functions are used all over the place, I see 15 different models use this specific call. In this case the coil index is not known, it's &returned. If you did call to the coil model to get the index, then you can directly access other data and these mining functions could all be depricated. I do like that idea. This is a seperate issue.